libsolidity/analysis/TypeChecker.cpp

```diff
@@ -709,6 +701,23 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 					m_errorReporter.typeError(6617_error, _identifier.location, "The suffixes _offset and _slot can only be used on non-constant storage variables.");
 					return size_t(-1);
 				}
+				else if (var && var->value() && !var->value()->annotation().type)
+				{
+					m_errorReporter.typeError(
+						2249_error,
+						_identifier.location,
+						"Constant variables cannot be forward referenced."
```

Comment:
```suggestion
						"Constant variables cannot be forward referenced from inline assembly."
```

Reply:
Actually if it is a literal, it should be fine.

Reply:
Isn't it a literal in this case?

Reply:
No, it is an expression